sklearn/ensemble/tests/test_forest.py

```diff
@@ -181,6 +185,48 @@ def test_regression(name, criterion):
     check_regression_criterion(name, criterion)
 
 
+def test_poisson_vs_mse():
+    rng = np.random.RandomState(42)
+    n_train, n_test, n_features = 500, 500, 10
+    X = datasets.make_low_rank_matrix(n_samples=n_train + n_test,
+                                      n_features=n_features, random_state=rng)
+
+    coef = rng.uniform(low=-2, high=2, size=n_features) / np.max(X, axis=0)
+    y = rng.poisson(lam=np.exp(X @ coef))
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=n_test,
+                                                        random_state=rng)
+
+    forest_poi = RandomForestRegressor(criterion="poisson",
+                                       n_estimators=10,
+                                       bootstrap=False,
+                                       random_state=rng)
+    forest_mse = RandomForestRegressor(criterion="squared_error",
+                                       n_estimators=10,
+                                       bootstrap=False,
+                                       random_state=rng)
+
+    forest_poi.fit(X_train, y_train)
+    forest_mse.fit(X_train, y_train)
+    dummy = DummyRegressor(strategy="mean").fit(X_train, y_train)
+
+    # Check for the balanced property before continuing
+    assert np.sum(forest_poi.predict(X_train)) == np.sum(y_train)
+    assert np.sum(forest_mse.predict(X_train)) == np.sum(y_train)
```

Comment:
I would prefer to have a separate test for this as is done in the tree module. Also, please use `pytest.approx` as some numerical rounding effects might cause a non-exact match.

Reply:
Ah I see, I've pushed this through